doc/nrf/libraries/others/nrf_compression.rst

```diff
@@ -0,0 +1,169 @@
+A unique ID for the compression library also needs to be defined, this should be globally available so that any applications or modules wanting to use this compression library can use it, as that is how compression libraries are indexed.
+
+Using the compression subsystem
+*******************************
+
+The following steps are required to decompress data using the nRF compression library (a flowchart diagram follows this description):
+
+* :c:func:`nrf_compress_implementation_find` should be used with the ID of the compression type to find, if this returns NULL then the compression type is not supported.
+* The ``init``, ``deinit`` and ``reset`` core function pointers should be checked, if any are NULL then there is an implementation problem with the compression type.
+* For decompression support the ``decompress_bytes_needed`` and ``decompress`` function pointers should be checked, if either are NULL then there is an implementation problem with the compression type.
+* The ``init`` function should be called to set the compression library up, if a non-zero error is returned then the compression library cannot be initialized, this could be due to lack of sufficient memory or other issues.
+* The ``decompress_bytes_needed`` function should be called to see how many bytes of data should be specified for decompression, if a non-zero error is returned then there is an issue with the compression library.
+* The ``decompress`` function should be called with the requested amount of data (or the maximum amount of data if there is less than that), if there is no more data to process then the ``last_part`` value should be set to true, otherwise false. if a non-zero error is returned then it is likely that the data provided to decompress is not valid.
+* If the ``offset`` updated value is not equal to the size of the input data provided then the next call to ``decompress`` should include the unused bytes from the buffer at the start of it.
+* If the ``output_size`` value is greater than 0 then the amount of data specified is available in the ``output`` buffer which should be copied or moved to its intended destination.
+* The process of calling ``decompress_bytes_needed`` followed by ``decompress`` should continue until all the data has been processed.
+* The ``deinit`` function should be called to set clean the compression library up.
```

Comment:
```suggestion
To decompress data using the nRF Compression library, complete the following steps:

1. Use the :c:func:`nrf_compress_implementation_find` function with the ID of the desired compression type. 
   If it returns ``NULL``, the compression type is not supported.
#. Check the ``init``, ``deinit`` and ``reset`` core function pointers.
   If any return ``NULL``, there is an implementation issue with the compression type.
#. For decompression support, check the :c:type:`nrf_compress_decompress_bytes_needed_t` and :c:func:`nrf_compress_decompress_func_t` function pointers. 
   If either is ``NULL``, there is an implementation issue with the compression type.
#. Call the :c:func:`nrf_compress_init_func_t` function to set up the compression library.
   If a non-zero error is returned, the library cannot be initialized, possibly due to insufficient memory or other issues.
#. Call the :c:type:`nrf_compress_decompress_bytes_needed_t` function to determine how many bytes of data should be specified for decompression.
   If a non-zero error is returned, there is an issue with the compression library.
#. Call the :c:func:`nrf_compress_decompress_func_t` function with the requested amount of data (or the maximum available, if less is available). 
   In case there is no more data to process, set the ``last_part`` value to true, otherwise set it to false. 
   If a non-zero error is returned, the provided data might be invalid.
#. Ensure that if the ``offset`` updated value is not equal to the size of the input data provided, the next call to  :c:func:`nrf_compress_decompress_func_t` function includes the unused bytes from the buffer at its start.
#. Verify if the ``output_size`` value is greater than ``0``.
   In such a case, the specified amount of data is available in the ``output`` buffer and you should copy it or move it to its intended destination.
#. Repeat the process of calling the :c:type:`nrf_compress_decompress_bytes_needed_t` function followed by  :c:func:`nrf_compress_decompress_func_t` until all the data has been processed.
#. Call the :c:func:`nrf_compress_deinit_func_t` function to clean up the compression library.
```